### DISCUSSION TITLE
spacy.load error decorative function

### DISCUSSION THREAD

**Discussion by AMS-L** [explosion/spaCy#13146] [Help: Installation]
## How to reproduce the behaviour
  !python3 -m spacy download en_core_web_sm  
import spacy
nlp = spacy.load("en_core_web_sm")

## Your Environment

2023-11-20 22:52:39.399591: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT

## Info about spaCy

- **spaCy version:** 3.5.0
- **Platform:** Linux-6.2.0-36-generic-x86_64-with-glibc2.35
- **Python version:** 3.10.12
- **Pipelines:** fr_core_news_sm (3.5.0), en_core_web_sm (3.5.0), fr_core_news_md (3.5.0)

* Python Version Used: 3.10.12
* spaCy Version Used: 3.5.0

TypeError                                 Traceback (most recent call last)
Cell In[33], line 1
----> 1 nlp = spacy.load("en_core_web_sm")

File [~/.local/lib/python3.10/site-packages/spacy/__init__.py:54](https://file+.vscode-resource.vscode-cdn.net/home/amine/Documents/S9/PFEE/PFEE_project2/research/~/.local/lib/python3.10/site-packages/spacy/__init__.py:54), in load(name, vocab, disable, enable, exclude, config)
     30 def load(
     31     name: Union[str, Path],
     32     *,
   (...)
     37     config: Union[Dict[str, Any], Config] = util.SimpleFrozenDict(),
     38 ) -> Language:
     39     """Load a spaCy model from an installed package or a local path.
     40 
     41     name (str): Package name or model path.
   (...)
     52     RETURNS (Language): The loaded nlp object.
     53     """
---> 54     return util.load_model(
     55         name,
     56         vocab=vocab,
     57         disable=disable,
     58         enable=enable,
     59         exclude=exclude,
     60         config=config,
...
    141 assert hasattr(self, 'name')
    142 if not hasattr(self, 'signature'):
--> 143     raise TypeError('You are decorating a non function: %s' % func)

TypeError: You are decorating a non function: that is odd

**Comment by svlandeg** [comment]
Hi, welcome to the forum!

We'd appreciate it if you'd use markdown for formatting code & output, as that makes it all more readable. You can surround a code block with three backticks.

This code seems to be just fine, what kind of environment are you executing this in? Was the model downloaded correctly, any error or warning output? Did you restart your kernel after downloading the model?

To be perfectly honest I'm not sure what's going on here - I've never seen this kind of error before and I've executed `spacy.load` a million times. My best guess is that some sort of corrupt model is being loaded, so I'd advice creating a clean environment, reinstalling spacy and trying this again. 

Let us know how you go?
